File: Domain/Event.cs

namespace Domain
{
    public class Event
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public int Capacity { get; set; }
        public string SocialMediaLink { get; set; }
        public string Type { get; set; }
    }
}

File: client/src/app/models/event.ts
import { User } from "."
export interface Event {
    id: string
    name: string
    description: string
    date: string
    city: string
    venue: string
    capacity: number
    socialMediaLink: string
    type: EventType
    guests: User[]
}

type EventType = 'Book club meeting' | 'Book launch party' | 'Literary Festival' | 'Author meet and greet';
